packages/stores/src/account/__tests_e2e__/positions.spec.ts

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable */
+import { ObservableQueryPool } from "src/queries";
+import {
+  chainId,
+  getLatestQueryPool,
+  RootStore,
+  waitAccountLoaded,
+} from "../../__tests_e2e__/test-env";
+import { maxTick, minTick } from "@osmosis-labs/math";
+// import { Int } from "@keplr-wallet/unit";
+
+describe("Create CL Positions Txs", () => {
+  const { accountStore, queriesStore, chainStore } = new RootStore();
+  let queryPool: ObservableQueryPool | undefined;
+
+  beforeAll(async () => {
+    const account = accountStore.getAccount(chainId);
+    account.cosmos.broadcastMode = "sync";
+    await waitAccountLoaded(account);
+  });
+
+  beforeEach(async () => {
+    const account = accountStore.getAccount(chainId);
+
+    // prepare CL pool
+    await new Promise((resolve, reject) =>
+      account.osmosis.sendCreateConcentratedPoolMsg(
+        "uion",
+        "uosmo",
+        1,
+        0.001, // must have spread factor to generate fees
+        undefined,
+        (tx) => {
+          if (tx.code) reject(tx.log);
+          else resolve(tx);
+        }
+      )
+    );
+
+    queryPool = await getLatestQueryPool(chainId, queriesStore);
+  });
+
+  it("should be able to be created", async () => {
+    await expect(createFullRangePosition(queryPool!.id)).resolves.toBeDefined();
+    await expect(
+      getUserPositionsIdsForPool(queryPool!.id)
+    ).resolves.toHaveLength(1);
+    await expect(createFullRangePosition(queryPool!.id)).resolves.toBeDefined();
+    await expect(
+      getUserPositionsIdsForPool(queryPool!.id)
+    ).resolves.toHaveLength(2);
+  });
+
+  it("can have liquidity be added to it", async () => {
+    const account = accountStore.getAccount(chainId);
+
+    // create 2 positions, since there need to be at least 2 to be able to add liquidity
+    await createFullRangePosition(queryPool!.id);
+    await createFullRangePosition(queryPool!.id);
+    const userPositionIds = await getUserPositionsIdsForPool(queryPool!.id);
+    const lastPositionId = userPositionIds[userPositionIds.length - 1];
+
+    // add liquidity to position
+    const specifiedAmount0 = "1000";
+    const calculatedAmount1 = "1000";
+
+    await expect(
+      new Promise((resolve, reject) =>
+        account.osmosis
+          .sendAddToConcentratedLiquidityPositionMsg(
+            lastPositionId,
+            specifiedAmount0,
+            calculatedAmount1,
+            undefined,
+            undefined,
+            (tx) => {
+              if (tx.code) reject(tx.log);
+              else resolve(tx);
+            }
+          )
+          .catch(reject)
+      )
+    ).resolves.toBeDefined();
+
+    // old position is replaced
+    await expect(
+      getUserPositionsIdsForPool(queryPool!.id)
+    ).resolves.toHaveLength(2);
```

Comment:
is it worth checking what the actual values are?

Reply:
The positions IDs may not be deterministically created depending on other tests in the env, so I don't think we can reliably specify the expected IDs. If other tests create positions and run before these tests, the IDs would change and break these tests.

Reply:
👍 Maybe just a comment would be good then?